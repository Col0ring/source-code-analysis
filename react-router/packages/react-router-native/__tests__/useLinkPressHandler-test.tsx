import * as React from "react";
import { Text, TouchableHighlight, View } from "react-native";
import type { LinkProps } from "react-router-native";
import {
  NativeRouter,
  Route,
  Routes,
  useLinkPressHandler
} from "react-router-native";
import * as TestRenderer from "react-test-renderer";

import { press } from "./utils";

describe("Custom link with useLinkPressHandler", () => {
  function Link({ to, replace, state, ...rest }: LinkProps) {
    let handlePress = useLinkPressHandler(to, { replace, state });
    return <TouchableHighlight {...rest} onPress={handlePress} />;
  }
  it("navigates to the new view", () => {
    function Home() {
      return (
        <View>
          <Text>Home</Text>
          <Link to="../about">
            <Text>About</Text>
          </Link>
        </View>
      );
    }

    function About() {
      return (
        <View>
          <Text>About</Text>
        </View>
      );
    }

    let renderer: TestRenderer.ReactTestRenderer;
    TestRenderer.act(() => {
      renderer = TestRenderer.create(
        <NativeRouter initialEntries={["/home"]}>
          <Routes>
            <Route path="home" element={<Home />} />
            <Route path="about" element={<About />} />
          </Routes>
        </NativeRouter>
      );
    });

    expect(renderer.toJSON()).toMatchSnapshot();

    let touchable = renderer.root.findByType(TouchableHighlight);

    TestRenderer.act(() => {
      press(touchable);
    });

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
