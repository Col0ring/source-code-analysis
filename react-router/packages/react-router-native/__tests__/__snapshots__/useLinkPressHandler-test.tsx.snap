// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom link with useLinkPressHandler navigates to the new view 1`] = `
<View>
  <Text>
    Home
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={null}
    >
      About
    </Text>
  </View>
</View>
`;

exports[`Custom link with useLinkPressHandler navigates to the new view 2`] = `
<View>
  <Text>
    About
  </Text>
</View>
`;
